K8s-deployment/Charts/file-server/values.yaml

```diff
@@ -305,23 +304,18 @@ apiServer:
   ## @param apiServer.extraVolumes Optionally specify extra list of additional volumes for the ApiServer pod(s)
   ##
   extraVolumes:
-    - name: fs-config-apiserver
+    - name: fs-config
       secret:
-        secretName: fs-config-apiserver
-    - name: fs-keystore
-      secret:
-        secretName: fs-keystore
+        secretName: fs-config
     - name: fs-data
       persistentVolumeClaim:
         claimName: fs-data
```

Comment:
Not needed in a Statefulset when a persistent volume claim template is being used.

Reply:
yes not required....but I kept it as it was in here
https://github.com/datakaveri/iudx-deployment/blob/a71774ec30305fac07164ef46793ff90a2672d62/K8s-deployment/file-server/fs-deployment/api-server.yaml#L66

Reply:
It is not optional to have it there. It is incorrect, you cannot have a `persistentVolumeClaim` under `volumes` in statefulsets.
The correct way is to use `persistentVolumeClaimTemplates` so that volumes are provisioned dynamically as the application is scaled. Please remove it.

Reply:
@abhi4578 should remove this from the api-server yaml as well.

Reply:
Thanks @karun-singh. its not needed to be defined in volumes as its already  defined in ```volumeClaimTemplates``` . Have removed it now in commit https://github.com/datakaveri/iudx-deployment/commit/72c9664aaebfab7478d48baf8a674a32a38f3f1d . Also there is no autoscaling enabled as of now on file server pods.

Reply:
no meaningful way to do simple autoscaling. it will ultimately depend on storage medium to transfer files/speed of transfer on demand. current assumption is the demand/no.of people requesting is not more and this should be sufficient. But this should be fine tuned accordingly at storage level and k8s level when the need come/demand for it increases.